src/dune_lang/dune_project.ml

```diff
@@ -953,7 +953,28 @@ let parse ~dir ~(lang : Lang.Instance.t) ~file =
          "warnings"
          ~default:Warning.Settings.empty
          (Syntax.since Stanza.syntax (3, 11) >>> Warning.Settings.decode)
+     and+ pkg_field =
+       field_o
+         "pkg"
+         (Syntax.since Stanza.syntax (3, 21)
+          >>> let+ loc = loc
+              and+ () = junk_everything in
+              loc)
      in
```

Comment:
No point putting a since here since it was never possible before. Instead I would just `>>|` the `field_o` above into `Option.iter` which raises the user error below. Then you don't need `pkg_field`, it would just become `()`.

Reply:
If a user is in a situation where they have put `pkg` in their `dune-project` file, then the version that they are on shouldn't really matter. This may prompt them to upgrade their `dune lang` version first and then mention that it should be in `dune-workspace`. I would rather we first mention that we are in `dune-workspace` and then the correct stanza would be arguing about the version constraint.

Reply:
The `pkg` stanza doesn't exist in `dune-project`, we are simply giving a helpful error message for a (possibly) common mistake.